aep/general/0135/aep.md.j2

```diff
@@ -0,0 +1,314 @@
+  resource state returns a consistent response.
+
+Delete methods implement a common request pattern:
+
+```proto
+message DeleteBookRequest {
+  // The name of the book to delete.
+  // Format: publishers/{publisher}/books/{book}
+  string name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.resource_reference) = {
+      type: "library.googleapis.com/Book"
+    }];
+}
+```
+
+- A `name` field **must** be included. It **should** be called `name`.
+  - The field **should** be [annotated as required][aep-203].
+  - The field **must** identify the [resource type][aep-123] that it
+    references.
+- The comment for the field **should** document the resource pattern.
+- The request message **must not** contain any other required fields, and
+  **should not** contain other optional fields except those described in this
+  or another AEP.
+
+{% endtabs %}
+
+### Soft delete
+
+**Note:** This material was moved into its own document to provide a more
+comprehensive treatment: [AEP-164][aep-164].
+
+### Long-running delete
+
+Some resources take longer to delete than is reasonable for a
+regular API request. In this situation, the API **should** use a long-running
+operation instead. See [AEP-151][aep-151] for full details on long-running operations.
+
+{% tab oas -%}
+
+{% sample 'long_running_delete.oas.yaml', 'paths' %}
+
+- The response status code should be `202 Accepted` if the request was
+  accepted for later processing. When the request is processed it could still fail.   
+- The `response` field of the response body **must** be set to the appropriate return type if the RPC
+  was not long-running: `google.protobuf.Empty` for most Delete RPCs, or the
+  resource itself for soft delete ([AEP-164][aep-164]).
+- Both the `response_type` and `metadata_type` fields **must** be specified
+  (even if they are `google.protobuf.Empty`).
+
+{% tab proto -%}
+
+```proto
+rpc DeleteBook(DeleteBookRequest) returns (google.longrunning.Operation) {
+  option (google.api.http) = {
+    delete: "/v1/{name=publishers/*/books/*}"
+  };
+  option (google.longrunning.operation_info) = {
+    response_type: "google.protobuf.Empty"
+    metadata_type: "OperationMetadata"
+  };
+}
+```
+
+- The response type **must** be set to the appropriate return type if the RPC
+  was not long-running: `google.protobuf.Empty` for most Delete RPCs, or the
+  resource itself for soft delete ([AEP-164][aep-164]).
+- Both the `response_type` and `metadata_type` fields **must** be specified
+  (even if they are `google.protobuf.Empty`).
+
+{% endtabs %}
+
+### Cascading delete
+
+Sometimes, it may be necessary for users to be able to delete a resource as
+well as all applicable child resources. However, since deletion is usually
+permanent, it is also important that users not do so accidentally, as
+reconstructing wiped-out child resources may be quite difficult.
+
+If an API allows deletion of a resource that may have child resources, the API
```

Comment:
I wonder if this guidance makes sense universally - I could imagine some resources would require child resources to be deleted along with them? e.g. a static IP resources on an router resource?

Reply:
Not sure if you are suggesting a change here, but this language comes straight from the Google AIP so I propose we keep it for now and consider changes in a future PR.